specs/ProcessInfo.md

```diff
@@ -0,0 +1,251 @@
+
+#undef KIND_ENTRY
+    }
+
+    return L"PROCESS KIND: " + std::to_wstring(static_cast<uint32_t>(kind));
+}
+
+// Get the process info
+//! [ProcessInfoChanged]
+void ProcessComponent::ProcessInfo()
+{
+    std::wstring result;
+    UINT process_list_size;
+    CHECK_FAILURE(m_processCollection->get_Count(&process_list_size));
+
+    if (process_list_size == 0)
+    {
+        result += L"No process found.";
+    }
+    else
+    {
+        result += std::to_wstring(process_list_size) + L" process(s) found";
+        result += L"\n\n";
+        for (UINT i = 0; i < process_list_size; ++i)
+        {
+            UINT32 processId = 0;
+            CHECK_FAILURE(
+                m_processCollection->GetProcessIdAtIndex(i, &processId));
+
+            WCHAR buffer[4096] = L"";
+            StringCchPrintf(buffer, ARRAYSIZE(buffer), L"Process ID: %u\n", processId);
+
+            result += buffer;
+
+            COREWEBVIEW2_PROCESS_KIND kind;
+            CHECK_FAILURE(m_processCollection->GetProcessTypeAtIndex(i, &kind));
+
+            result += L"Process Kind: " + ProcessKindToString(kind);
+            result += L"\n";
+        }
+    }
+    MessageBox(nullptr, result.c_str(), L"GetProcessesInfo Result", MB_OK);
+}
+//! [ProcessInfoChanged]
+```
+
+# API Details    
+```
+interface ICoreWebView2StagingEnvironment5;
+interface ICoreWebView2StagingProcessCollection;
+interface ICoreWebView2StagingProcessInfoChangedEventHandler;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_KIND {
+  /// Indicates the browser process kind.
+  COREWEBVIEW2_PROCESS_KIND_BROWSER,
+
+  /// Indicates the render process kind.
+  COREWEBVIEW2_PROCESS_KIND_RENDERER,
+
+  /// Indicates the utility process kind.
+  COREWEBVIEW2_PROCESS_KIND_UTILITY,
+
+  /// Indicates the sandbox helper process kind.
+  COREWEBVIEW2_PROCESS_KIND_SANDBOX_HELPER,
+
+  /// Indicates the GPU process kind.
+  COREWEBVIEW2_PROCESS_KIND_GPU,
+
+  /// Indicates the PPAPI plugin process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_PLUGIN,
+
+  /// Indicates the PPAPI plugin broker process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER,
+
+  /// Indicates the process of unspecified kind.
+  COREWEBVIEW2_PROCESS_KIND_UNKNOWN,
+} COREWEBVIEW2_PROCESS_KIND;
+
+[uuid(20856F87-256B-41BE-BD64-AB1C36E3D944), object, pointer_default(unique)]
+interface ICoreWebView2StagingEnvironment5 : IUnknown
+{
+  /// Adds an event handler for the `ProcessInfoChanged` event.
+  /// 
+  /// \snippet ProcessComponent.cpp ProcessInfoChanged
+  HRESULT add_ProcessInfoChanged(
+      [in] ICoreWebView2StagingProcessInfoChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ProcessInfoChanged`.
+  HRESULT remove_ProcessInfoChanged(
+      [in] EventRegistrationToken token);
+
+  /// Returns the `ICoreWebView2StagingProcessCollection`
+  [propget] HRESULT ProcessInfo([out, retval]ICoreWebView2StagingProcessCollection** value);
```

Comment:
How up to date is our process info? Does this need to be an async method instead of a property? Or do we always have info on the processes?

Reply:
Every time a new process being connected/disconnected, we notify the client side to add that process Id/Type to the existing list. (You can refer to my PR on the client side implementation).

Reply:
Should I still make this a complete handler like the `ICoreWebView2GetCookiesCompletedHandler`? But the list should be changing every time navigation happens. Now thinking about it, the `add_ProcessInfo` might not be needed. We can listen to the navigation complete event if process info got changed?